src/mongo/db/pipeline/expression.h

```diff
@@ -1219,17 +1219,23 @@ class ExpressionIn final : public ExpressionFixedArity<ExpressionIn, 2> {
 };
 
 
-class ExpressionIndexOfArray  : public ExpressionRangedArity<ExpressionIndexOfArray, 2, 4> {
+class ExpressionIndexOfArray : public ExpressionRangedArity<ExpressionIndexOfArray, 2, 4> {
 public:
     explicit ExpressionIndexOfArray(const boost::intrusive_ptr<ExpressionContext>& expCtx)
         : ExpressionRangedArity<ExpressionIndexOfArray, 2, 4>(expCtx) {}
 
-    Value evaluate(const Document& root) const ;
-    std::vector<Value> parseDeps(const Document& root,
-                                  const ExpressionVector& operands,
-                                  size_t arrayLength) const;
+    Value evaluate(const Document& root) const;
     boost::intrusive_ptr<Expression> optimize() final;
     const char* getOpName() const final;
+
+protected:
+    // Evaluates and validates the value that is being searched for and if given the start index and
+    // the end index passed to ExpressionIndexOfArray returns arguments in the form of a
+    // vector<Value>
```

Comment:
```
/**
 * Use this syntax for comments documenting the behavior of a function or method.
 */
```

Reply:
Also - can we re-word this comment to make it more clear what the vector of values will be? How about something like
`When given 'operands' which correspond to the arguments to $indexOfArray, evaluates and validates the target value, starting index, and ending index arguments and returns their values in that order. The starting index and ending index are optional, so the returned vector will have a length between 1 and 3. Throws a UserException if the values are found to be invalid in some way, e.g. if the indexes are not numbers.`